### PR TITLE
arm64 purecap kernel panic fixes

### PR DESCRIPTION
**Pull request by bsdjhb**
- arm64 db_trace: Use cheri_can_access for trapframes.
- arm64: Fix the pseudo FP (c29) for kernel exceptions for purecap.
- arm64 makectx: Fix overflow of tf_x array

### CODE REVIEW

**Comment by bsdjhb** [comment]
The third commit I'll merge upstream as well.

File: sys/arm64/arm64/db_trace.c [hunk 1]
```diff
@@ -94,6 +94,14 @@ db_stack_trace_cmd(struct thread *td, struct unwind_state *frame)
 			struct trapframe *tf;
 
 			tf = (struct trapframe *)(uintptr_t)frame->fp - 1;
+#ifdef __CHERI_PURE_CAPABILITY__
+			if (!cheri_can_access(tf,
+			    CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP,
+			    (vm_offset_t)tf, sizeof(*tf))) {
```
Comment (jrtc27) [inline sys/arm64/arm64/db_trace.c:100]: s/vm_offset_t/ptraddr_t/

Same underlying type and meaning but the prototype does say ptraddr_t for cheri_can_access

File: sys/arm64/arm64/db_trace.c [hunk 2]
```diff
@@ -94,6 +94,14 @@ db_stack_trace_cmd(struct thread *td, struct unwind_state *frame)
...
+				db_printf("--- invalid trapframe %#p\n", tf);
+				break;
+			}
+#endif
 			if (!kstack_contains(td, (vm_offset_t)tf,
```
Comment (jrtc27) [inline sys/arm64/arm64/db_trace.c:105]: I wonder if we should be checking alignment too upstream and downstream like we do in unwind_frame?

File: sys/arm64/arm64/machdep.c [hunk 3]
```diff
@@ -380,8 +380,11 @@ makectx(struct trapframe *tf, struct pcb *pcb)
 {
 	int i;
 
-	for (i = 0; i < nitems(pcb->pcb_x); i++)
+	for (i = 0; i < nitems(pcb->pcb_x); i++) {
+		if (i == PCB_LR)
+			continue;
```
Comment (jrtc27) [inline sys/arm64/arm64/machdep.c:385]: Could move the tf->tf_elr read in here?

File: sys/arm64/arm64/db_trace.c (hunk 2 above)
Comment (bsdjhb) [reply sys/arm64/arm64/db_trace.c:105]: Yes, I think we should add the alignment check upstream.

**Pull request merged**